src/reciperepository.js

```diff
@@ -0,0 +1,34 @@
+class RecipeRepository {
+  constructor(recipes) {
+    this.recipes = recipes
+
+  }
+
+  filterRecipesViaTags(tags) {
+    const search =  this.recipes.reduce((recipes, recipe) => {
+      let count = 0;
+      tags.forEach(tag => {
+        if (recipe.tags.includes(tag)) {
+          count += 1;
+        }
+      })
+      if (count === tags.length) {
+        recipes.push(recipe.name)
+      }
+      return recipes
+    }, [])
+    return search
+  }
```

Comment:
Changed this method to take in an array of tags. each loop through the recipes array (so each recipe) now counts how many tags it matches. At the end of the loop iteration if that count is equal to the number of tags in the argument array it pushes to the return array.

It's a lot clunkier than what you started with but I think making the code more dynamic will benefit us in the long run.

Reply:
This change looks great!